kensu/utils/rule_engine.py

```diff
@@ -0,0 +1,166 @@
+from kensu.utils.kensu_provider import KensuProvider
+import warnings
+
+def add_rule(data_source, field, type, parameters):
+    k = KensuProvider().instance()
+    k.rules.append({data_source:
+                        {'field': field,
+                         'fun': {"name": type, "arguments": parameters}
+                         }})
+
+def add_min_max(data_source, field, min = None, max = None):
+    parameters = {}
+    if min is not None:
+        parameters["minVal"] = min
+    if max is not None:
+        parameters["maxVal"] = max
+    add_rule(data_source,field,type='Range',parameters=parameters)
+
+def add_missing_value_rules(data_source, data_frame):
+    for col in data_frame.columns:
+        field = col+'.nullrows'
+        add_rule(data_source, field, type='Range', parameters={'maxVal':0})
+
+def add_frequency_rule(data_source, hours = None, days = None, weeks = None, months = None):
+
+    if hours:
+        timeLapseUnit = 'Hours'
+        timeLapse = hours
+    elif days:
+        timeLapseUnit = 'Days'
+        timeLapse = days
+    elif weeks:
+        timeLapseUnit = 'Weeks'
+        timeLapse = weeks
+    elif months:
+        timeLapseUnit = 'Months'
+        timeLapse = months
+
+    add_rule(data_source,field=None,
+             type='Frequency',
+             parameters={'timeLapse': timeLapse,'timeLapseUnit':timeLapseUnit})
+
+
+#TODO Variabilty over time
+def add_variability_rule(data_source, variation, hours = None, days = None, weeks = None, months = None):
+
+    parameters = {}
+    parameters['variation'] = variation
+
+    add_rule(data_source,field=None,
+             type='Variability',
+             parameters=parameters)
+
+def check_format_consistency(data_source):
+    k = KensuProvider().instance()
+
+    from kensu.sdk import get_latest_datasource_in_logical
+    cookie = k.get_cookie()
+    checked_format = get_latest_datasource_in_logical(k.api_url.replace("-api",""),cookie,data_source)['format']
+    previous_ds = get_latest_datasource_in_logical(k.api_url.replace("-api",""),cookie,data_source,n=-2)
```

Comment:
keep in mind this and some other rule checks might be be impacted by delay between entities sent to kensu and getting actually ingested? i.e. when retrieving stuff from APIs we see a snapshot of let say few ~ minutes ago.

are there cases where this would cause issues (or false assumptios? e.g. DS not found yet, because was ingested only seconds ago) @SammyEK